core/src/test/java/org/elasticsearch/search/suggest/SuggestBuilderTests.java

```diff
@@ -60,6 +131,31 @@ protected NamedWriteableRegistry provideNamedWritableRegistry() {
     }
 
     /**
+     * Test that a malformed JSON completion suggestion request fails.
+     */
+    public void testMalformedJsonRequestPayload() throws Exception {
+        final String field = RandomStrings.randomAsciiOfLength(getRandom(), 10).toLowerCase(Locale.ROOT);
+        final XContentBuilder suggest = jsonBuilder().startObject()
+                                            .startObject("bad-payload")
+                                            .field("prefix", "sug")
+                                            .startObject("completion")
+                                            .field("field", field)
+                                            .startArray("payload")
+                                            .startObject()
+                                            .field("payload", "field")
+                                            .endObject()
+                                            .endArray()
+                                            .endObject()
+                                            .endObject().endObject();
+        try {
+            final SuggestBuilder suggestBuilder = SuggestBuilder.fromXContent(newParseContext(suggest), suggesters);
+            fail("Should not have been able to create SuggestBuilder from malformed JSON: " + suggestBuilder);
+        } catch (IllegalArgumentException e) {
+            assertThat(e.getMessage(), equalTo("Suggestion must have name"));
```

Comment:
The original test was testing something different in the sense that it had to send a request with the JSON payload in order for the shards to parse them and respond back with a parse exception.  Now, we are using the builders internally instead of raw bytes, so the "validation" happens much earlier.  In this case, its more of a unit test on the validity of the JSON, so that's why the failure reason has also changed.